x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/detail/hooks/use_change_log.ts

```diff
@@ -8,24 +8,23 @@ import { useMemo } from 'react';
 
 import { useGetFileByPathQuery } from '../../../../../hooks';
 import { getBreakingChanges, parseYamlChangelog } from '../utils';
+import { type PackageInfo } from '../../../../../types';
 
 /**
  * @param packageName the package to get the changelog for
  * @param latestVersion the version of changelog for the specified package
  * @param currentVersion is used to display the changelog starting from this version up to the latest version
  */
-export const useChangelog = (
-  packageName: string,
-  latestVersion: string,
-  currentVersion?: string
-) => {
-  const { data, error, isLoading } = useGetFileByPathQuery(
-    `/package/${packageName}/${latestVersion}/changelog.yml`
-  );
+export const useChangelog = (packageInfo: PackageInfo, currentVersion?: string) => {
+  const path = 'path' in packageInfo ? packageInfo.path : null;
```

Comment:
I think packageInfo should have an array of path in paths I think we could check if that array contains a changelog

Reply:
Actually looks like paths it's something we only have for installed package, we could probably catch 404 here

Reply:
Gotcha, I'll rework it to:
* Construct the path optimistically from latestVersion and packageName and try to fetch the changelog (as before)
* Handle 404 errors as an empty changelog with no error, otherwise return all other errors